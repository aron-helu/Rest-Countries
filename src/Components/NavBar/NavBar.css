
.navigation-bar {
  display: flex;
  justify-content: space-between;
  height: 50px;
  align-items: center;
  padding: 5px 15px;
  background-color: #18395d;
  background-image: url('../../image/dark-mosaic.png');
}

.ar-img {
  height: 30px;
  aspect-ratio: 1 / 1;
}

.logo-p {
  display: flex;
  align-items: center;
  text-align: center;
  font-size: 0.99rem;
}

.logo-div {
  display: flex;
  gap: 10px;
  align-items: center;
}

.NavLink {
  font-size: 0.95rem;
  list-style-type: none;

}

@media all and (min-width: 768px) {
  .navigation-bar {
    padding-top: 6px 10px;
    height: 70px;
  }
  .ar-img {
    height: 50px;
  }
  .NavLink {
    font-size: 1.4em;
    display: flex;
    align-items: center;
    margin-top: 10px;
  }

  .logo-p {
    font-size: 2.1em;
  }
}