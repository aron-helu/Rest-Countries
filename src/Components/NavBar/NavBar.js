import { NavLink, useLocation } from "react-router-dom";
import aron from '../../image/aron-transparent.png';
import './NavBar.css';

const NavBar = () => {
    const location = useLocation();
    return (
        <nav className="navigation-bar">
            <ul className="navigation-ul">
                <li className="NavLink">
                    <NavLink to="/" className="log">
                        {location.pathname === '/' ? 'COUNTRY' : ' <  BACK ' }
                    </NavLink>
                </li>
            </ul>

            <div className="logo-div">
                <img className="ar-img" src={aron} alt="world" />
                <p className="logo-p">Where in the world?</p>
            </div>
        </nav>
    );
}

export default NavBar;