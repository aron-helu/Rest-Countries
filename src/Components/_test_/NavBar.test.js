import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import React from 'react';
import NavBar from '../NavBar/NavBar';

test('snapshot for App', () => {
    const tree = render(
        <React.StrictMode>
            <Provider store={store}>
                <BrowserRouter>
                    <NavBar />
                </BrowserRouter>
            </Provider>
        </React.StrictMode>
    );
    expect(tree).toMatchSnapshot();
});