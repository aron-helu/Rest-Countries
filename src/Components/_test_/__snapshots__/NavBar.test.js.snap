// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navigation-bar"
      >
        <ul
          class="navigation-ul"
        >
          <li
            class="NavLink"
          >
            <a
              aria-current="page"
              class="log active"
              href="/"
            >
              COUNTRY
            </a>
          </li>
        </ul>
        <div
          class="logo-div"
        >
          <img
            alt="world"
            class="ar-img"
            src="aron-transparent.png"
          />
          <p
            class="logo-p"
          >
            Where in the world?
          </p>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navigation-bar"
    >
      <ul
        class="navigation-ul"
      >
        <li
          class="NavLink"
        >
          <a
            aria-current="page"
            class="log active"
            href="/"
          >
            COUNTRY
          </a>
        </li>
      </ul>
      <div
        class="logo-div"
      >
        <img
          alt="world"
          class="ar-img"
          src="aron-transparent.png"
        />
        <p
          class="logo-p"
        >
          Where in the world?
        </p>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
